go-controller/pkg/ovn/controller/networkconnect/topology.go

```diff
@@ -0,0 +1,913 @@
+		successfullyDeleted = append(successfullyDeleted, owner)
+	}
+
+	// Execute all operations (create and delete) in a single transaction
+	if len(ops) > 0 {
+		if _, err := libovsdbops.TransactAndCheck(c.nbClient, ops); err != nil {
+			errs = append(errs, fmt.Errorf("failed to execute network connection operations for CNC %s: %w", cncName, err))
+			return utilerrors.Join(errs...)
+		}
+		klog.Infof("CNC %s: executed %d sync network connections operations", cncName, len(ops))
+	}
+
+	// Update cache only after successful transact - this ensures cache stays consistent with OVN NB
+	for _, added := range successfullyAdded {
+		cncState.connectedNetworks.Insert(added.owner)
+		cncState.connectedNetworksRouterNames[added.owner] = added.networkRouterName
+	}
+	for _, deleted := range successfullyDeleted {
+		cncState.connectedNetworks.Delete(deleted)
+		delete(cncState.connectedNetworksRouterNames, deleted)
+	}
+
+	return utilerrors.Join(errs...)
+}
+
+// cleanupNetworkConnections removes all network connections for a CNC.
+// This is called when a CNC is being deleted. The ops are ordered to avoid referential integrity violations:
+// 1. First delete connect router ports (which have peer references to network router ports)
+// 2. Then delete network router ports
+// 3. Finally delete routing policies
+func (c *Controller) cleanupNetworkConnections(cncName string, cncState *networkConnectState) error {
+	var ops []ovsdb.Operation
+	connectRouterName := getConnectRouterName(cncName)
+
+	// For each connected network, clean up ports and policies on the routers.
+	// The connect router port has `peer` pointing to the network router port, so we must delete
+	// the connect router port first (or the network router port deletion will fail).
+	for owner, networkRouterName := range cncState.connectedNetworksRouterNames {
+		_, networkID, parseErr := parseOwnerKey(owner)
+		if parseErr != nil {
+			klog.Warningf("Failed to parse owner key %s during CNC cleanup: %v", owner, parseErr)
+			continue
+		}
+
+		// 1. Delete connect router ports for this network first (has peer reference to network router port)
+		var err error
+		ops, err = libovsdbops.DeleteLogicalRouterPortWithPredicateOps(c.nbClient, ops, connectRouterName, func(item *nbdb.LogicalRouterPort) bool {
+			return item.ExternalIDs[libovsdbops.ObjectNameKey.String()] == cncName &&
+				item.ExternalIDs[libovsdbops.NetworkIDKey.String()] == strconv.Itoa(networkID)
+		})
+		if err != nil {
+			return fmt.Errorf("failed to delete connect router ports for network %s: %w", owner, err)
+		}
+
+		// 2. Delete network router ports created by this CNC
+		ops, err = libovsdbops.DeleteLogicalRouterPortWithPredicateOps(c.nbClient, ops, networkRouterName, func(item *nbdb.LogicalRouterPort) bool {
+			return item.ExternalIDs[libovsdbops.ObjectNameKey.String()] == cncName &&
+				item.ExternalIDs[libovsdbops.NetworkIDKey.String()] == strconv.Itoa(networkID)
```

Comment:
is the networkIDKey match does anything extra? can't we just remove by using cnc name?